Improve how Pending Refund payments are displayed

erdemyerebasmaz opened the pull request:
This PR updates the UI for pending refunds & completed refunds.

### Changelist:
- Pending refund items now have a "Pending Refund" label & uses a refund a icon as their avatar.
- Pending refund & completed refund items where we can't track the exact transaction amount now display their total estimated refundable amount(_payment amount + fee_) as unsigned on Amount field on the payment list, and in refund tx amount on their detail view.

#### Screenshots:
##### Pending Refunds
<img width="40%" alt="image" src="https://github.com/user-attachments/assets/8cf48a08-b293-4cb6-8e90-bfbf104c0131" />

##### Completed Refunds _where we can't track the exact transaction amount (`isRefunded && refundTxAmountSat == 0`)_
<img width="40%" alt="image" src="https://github.com/user-attachments/assets/1a95d629-57ed-48da-9a0a-84ad0510124c" /> <img width="40%" alt="image" src="https://github.com/user-attachments/assets/b3c1ec34-4f7c-499d-a981-1bc3722f1079" />

dangeross commented on lib/routes/home/widgets/payments_list/widgets/payment_item/widgets/payment_item_amount.dart:
```diff
@@ -23,19 +23,31 @@ class PaymentItemAmount extends StatelessWidget {
                 includeDisplayName: false,
               );
 
+              if (paymentData.status == PaymentState.refundPending) {
+                amount = currencyState.bitcoinCurrency.format(
+                  paymentData.amountSat + actualFeeSat,
+                  includeDisplayName: false,
+                );
+
+                actualFeeSat = 0;
+              }
```
Is there a reason this is changed only during the pending refund? It will just go back to the original values when the refund is complete

erdemyerebasmaz replied:
Addressed with:
- [Display total refundable amount on Amount widget](https://github.com/breez/misty-breez/pull/410/commits/a2a745f7a1e7ebc74a857200baaa8c32aafa2eb3)

where we display the estimated total refund amount (payment amount + fees) for:

1. pending refunds (PaymentState.refundPending)
2. completed refunds where we can't track the exact transaction amount (isRefunded && refundTxAmountSat == 0)

dangeross replied:
Ok, I think this will only show the correct amount for amountless receive chain swaps. If an amount was selected to be received it show the expected payable amount (not actual send amount)

erdemyerebasmaz replied:
Refund swaps have their amount fixed, we don't allow the user to change it on Misty Breez.

Note: Amountless swaps are not integrated yet on Misty Breez.

erdemyerebasmaz replied:
That wasn't my experience, perhaps we update the payment data?

```
{
  "event": "update",
  "channel": "swap.update",
  "args": [
    {
      "id": "gh2XWg6yMouD",
      "status": "transaction.lockupFailed",
      "failureReason": "locked 26000 is more than expected 25000",
      "failureDetails": {
        "expected": 25000,
        "actual": 26000
      }
    }
  ],
  "timestamp": "1741297502609"
}
```

<img width="414" alt="image" src="https://github.com/user-attachments/assets/55dd5715-0fef-42f7-8c09-d7de2050d7e7" />

erdemyerebasmaz replied:
The behavior on the example I gave was due to a bug, which got addressed in 
- https://github.com/breez/breez-sdk-liquid/pull/771

Although the  refundable amount estimation is incorrect, we are keeping these changes as it's still an improvement over previous implementation and if i'm not mistaken, it will reflect the expected amount for the failed transaction.

erdemyerebasmaz commented on lib/routes/home/widgets/payments_list/widgets/payment_details_sheet/widgets/payment_details_sheet_refund_tx_fee_amount.dart:
great catch, resolved:
[rename PaymentDetailsSheetRefundTxAmount file for consistency](https://github.com/breez/misty-breez/pull/410/commits/f2b54c51529a41dc792bb869b90b9e85a36e1313)

erdemyerebasmaz commented on lib/routes/home/widgets/payments_list/widgets/payment_details_sheet/payment_details_sheet.dart:
```diff
@@ -158,20 +158,28 @@ class PaymentDetailsSheet extends StatelessWidget {
+                          paymentData: paymentData,
+                          labelAutoSizeGroup: _labelGroup,
+                        ),
+                      ],
```
No, it was never the case. There was a widget duplication for different refund states in case we decided to handle things differently.

I'd suggest taking a look at for the updated version:
- [ui: display estimated refund amount for pending & completed refunds w…](https://github.com/breez/misty-breez/pull/410/commits/b4cbc79c42a9f0296599d822deda5768fd791c32)

where we display the stimated total refund amount (original payment + fees) for:
1. pending refunds (`PaymentState.refundPending`)
2. completed refunds where we can't track the exact transaction amount (`isRefunded && refundTxAmountSat == 0`)

erdemyerebasmaz commented:
### Changelist:
- Show home page warning only when unrefunded swaps exist afe3956
  - Added `hasNonRefunded` getter to check for unrefunded swaps
- Display estimated refundable amounts for pending/completed refunds b4cbc79  a2a745f
  - Show original tx fee for completed refunds without refund tx amount
- Renamed `PaymentDetailsSheetRefundTxAmount` file for consistency f2b54c5

danielgranhao reviewed (commented):
A couple of nits

erdemyerebasmaz commented on lib/routes/home/widgets/payments_list/widgets/payment_details_sheet/widgets/payment_details_sheet_fee.dart:
```diff
@@ -42,9 +42,16 @@ class PaymentDetailsSheetFee extends StatelessWidget {
+
+                if (paymentData.isRefunded && paymentData.refundTxAmountSat == 0) {
+                  actualFeeSat = paymentData.feeSat;
+                }
```
Done a4823fa792b83a6d08b5c0a7e787f39e6882413b

erdemyerebasmaz commented on lib/routes/refund/pages/get_refund/widgets/refund_item_card/refund_item_card.dart:
```diff
@@ -20,6 +23,7 @@ class RefundItemCard extends StatelessWidget {
     final ThemeData themeData = Theme.of(context);
 
     final String lastRefundTxId = refundableSwap.lastRefundTxId ?? '';
+    final String txId = refundableSwap.lastRefundTxId ?? refundableSwap.swapAddress;
```
Is `refundId` a good alternative?
9e082b00d40bb53e91746e492e7bf029deb2fed8

Should we prevent opening `mempool.space` explorer on clicking `RefundableSwap.swapAddress` value?

danielgranhao replied:
Same concern expressed [here](https://github.com/breez/misty-breez/pull/418#discussion_r1991515667)

danielgranhao reviewed (commented):
Other than @dangeross's [concern](https://github.com/breez/misty-breez/pull/410#discussion_r1991445490), LGTM

erdemyerebasmaz commented:
Decided to merge current implementation and update it later once:
- https://github.com/breez/breez-sdk-liquid/issues/773

is addressed on SDK.
